src/tracer/tracee_memory.rs

```diff
@@ -274,6 +311,23 @@ mod poking {
             )
         }
 
+        #[test]
+        fn roundtrip_for_four_byte_word_doesnt_clobber_adjacent_four_bytes() -> R<()> {
+            run_roundtrip_test(|child, registers| {
+                let eight_bytes: [u8; 8] = [1, 2, 3, 4, 5, 6, 7, 8];
+                let four_bytes: [u8; 4] = [40, 30, 20, 10];
+
+                pokedata(child, registers.rdi, cast_to_eight_byte_word(eight_bytes))?;
+                poke_four_bytes(child, registers.rdi, cast_to_four_byte_word(four_bytes))?;
+
+                assert_eq!(
+                    cast_to_eight_byte_array(peekdata(child, registers.rdi)?),
+                    [40, 30, 20, 10, 5, 6, 7, 8],
+                );
+                Ok(())
+            })
+        }
+
```

Comment:
The next test below is a test for the test helper `run_roundtrip_test`. So I think the order of that one and your new test should be switched.